### DISCUSSION TITLE
How to mock test aiohttp.FormData?

### DISCUSSION THREAD

**Discussion by rahulnegi20** [aio-libs/aiohttp#7218] [Q&A]
https://stackoverflow.com/questions/75499668/how-to-mock-a-function-which-takes-http-request-as-argument-python

basically this, 
I want to test __call__() which takes http_request.
please let me know if any more information is required.
Thanks

**Comment by Dreamsorcerer** [comment]
This? https://docs.aiohttp.org/en/stable/testing.html#faking-request-object

Otherwise, I don't understand what you are trying to do. I don't see the relation between your first piece of code and the second..

**Reply by rahulnegi20** [reply]
> This? https://docs.aiohttp.org/en/stable/testing.html#faking-request-object
> 
> Otherwise, I don't understand what you are trying to do. I don't see the relation between your first piece of code and the second..

Can i send aiohttp.FormData() object through **`make_mocked_request()`** ?

like I did in my consumeAPI.py file.

API.py is server-side code serving this API call under this `__call__()` function. 

For now, I did build my own class that serves the FormData object like HTTP_req data. 

If still not clear let me know will present in much detail.

Thanks
